src/sciline/pipeline.py

```diff
@@ -572,15 +572,26 @@ def _get_provider(
                 typevars_in_expression = _extract_typevars_from_annotations(origin)
                 if typevars_in_expression:
                     explanation = [
-                        'Note that the type variables in',
-                        keyname(origin[tuple(typevars_in_expression)]),
-                        'have constraints',
-                        str(
-                            {
-                                keyname(tv): list(map(keyname, tv.__constraints__))
-                                for tv in typevars_in_expression
-                            }
-                        ),
+                        ''.join(
+                            map(
+                                str,
+                                (
+                                    'Note that the type variables ',
+                                    tuple(map(keyname, typevars_in_expression)),
+                                    ' of ',
+                                    keyname(origin),
+                                    ' have constraints ',
+                                    (
+                                        {
+                                            keyname(tv): tuple(
+                                                map(keyname, tv.__constraints__)
+                                            )
+                                            for tv in typevars_in_expression
+                                        }
+                                    ),
```

Comment:
This adds an note to the error message that looks something like this:
```python
sciline.handler.UnsatisfiedRequirement: ('No provider found for type', module.N[module.M[str], float], "Note that the type variables ('~T', '~S') of module.N have constraints {'~T': ('int', 'float'), '~S': ('M[int]', 'M[float]')}")
```
It's not very elegant. 
I think it somewhat obscures the real error message which is that no provider was found for the requested type. But it does kind of give you the information you need to understand what went wrong. So maybe it's still useful.

Reply:
A bit better, now it says:
```python
sciline.handler.UnsatisfiedRequirement: ('No provider found for type', module.N[module.M[str], float], "Note that N[~S, ~T] has constraints {'~S': ('M[int]', 'M[float]'), '~T': ('int', 'float')}")
```